doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,329 @@
+
+There is one mandatory Kconfig file in the SoC directory, and two optional
+files for a SoC:
+
+.. code-block:: none
+
+   soc/<vendor>/<your soc>
+   ├── Kconfig.soc
+   ├── Kconfig
+   └── Kconfig.defconfig
+
+:file:`Kconfig.soc`
+  A shared Kconfig file which can be sourced both in Zephyr Kconfig and sysbuild
+  Kconfig trees.
+
+  This file selects the SoC family and series in the Kconfig tree and potential
+  other SoC related Kconfig settings. In some cases a SOC_PART_NUMBER.
+  This file must not select anything outside the re-usable Kconfig SoC tree.
+
+  A :file:`Kconfig.soc` may look like this:
+
+  .. code-block:: kconfig
+
+     config SOC_<series name>
+             bool
+
+     config SOC_<SOC_NAME>
+             bool
+             select SOC_SERIES_<series name>
+
+     config SOC
+             default "SoC name" if SOC_<SOC_NAME>
+
+  Notice that ``SOC_NAME`` is a pure upper case version of the SoC name.
+
+  The Kconfig ``SOC`` setting is globally defined as a string and therefore the
+  :file:`Kconfig.soc` file shall only define the default string value and not
+  the type. Notice that the string value must match the SoC name used in the
+  :file:`soc.yml` file.
+
+:file:`Kconfig`
+  Included by :zephyr_file:`soc/Kconfig`.
+
+  This file can add Kconfig settings which are specific to the current SoC.
+
+  The :file:`Kconfig` will often indicate given hardware support using a setting
+  of the form ``HAS_<support>``.
+
+  .. code-block:: kconfig
+
+     config SOC_<SOC_NAME>
+             select ARM
+             select CPU_HAS_FPU
+
+  If the setting name is identical to an existing Kconfig setting in Zephyr and
+  only modifies the default value of said setting, then
+  :file:`Kconfig.defconfig` should be used  instead.
+
+:file:`Kconfig.defconfig`
+  SoC specific default values for Kconfig options.
+
+  Not all SoCs have a :file:`Kconfig.defconfig` file.
+
+  The entire file should be inside a pair of ``if SOC_<SOC_NAME>`` / ``endif``
+  or ``if SOC_SERIES_<SERIES_NAME>`` / ``endif``, like this:
+
+  .. code-block:: kconfig
+
+     if SOC_<SOC_NAME>
+
+     config NUM_IRQS
+             default 32
+
+     endif # SOC_<SOC_NAME>
+
+Multiple CPU clusters
+=====================
+
+CPU clusters must provide additional Kconfig settings in the :file:`Kconfig.soc`
+file. This will usually be in the form of ``SOC_<SOC_NAME>_<CLUSTER>`` so for
+a given ``soc1`` with two clusters ``clusterA`` and ``clusterB``, then this
+will look like:
+
+SoC's When a SoC defines CPU cluster
+
+  .. code-block:: kconfig
+
+     config SOC_SOC1_CLUSTERA
+             bool
+             select SOC_SOC1
+
+     config SOC_SOC1_CLUSTERB
+             bool
+             select SOC_SOC1
+
+
+.. _soc_porting_general_recommendations:
+
+General recommendations
+***********************
+
+Placeholder for time being.
+Do we have any general recommendations for SoC porting, like we have for Board
+porting, :ref:`porting-general-recommendations`.
```

Comment:
seems like this should be removed?

Reply:
Removed.

No one gave feedback on any recommendations, so guess that means we don't have any atm.